### DISCUSSION TITLE
Unable to delete pod stuck in terminating state

### DISCUSSION THREAD

**Discussion by BeyondEvil** [argoproj/argo-cd#10400] [Q&A]
Not sure if it's a bug or if I'm doing something wrong.

But we have a pod in ArgoCD that's been stuck in "terminating" state for two days. Trying to force-delete gives below error.

The pod does indeed not exist in the cluster when checking with `kubectl get pods`.
![Screenshot 2022-08-19 at 10 41 50](https://user-images.githubusercontent.com/15611634/185581069-13f70ca9-eb1c-4a0d-802d-5c305da538ba.png)
![Screenshot 2022-08-19 at 10 42 12](https://user-images.githubusercontent.com/15611634/185581089-34d56805-b824-490a-abcc-39afcc8ef7d9.png)

What can I do?

**Comment by zhaoxiaomin1983** [comment]
I met the same issue. How can I do to fix it?

**Comment by Babbili** [comment]
delete it this way
```bash
kubectl delete pod/POD_NAME --force --grace-period=0
```

**Reply by BeyondEvil** [reply]
I don't know about @zhaoxiaomin1983 but in my case that wouldn't have worked (I guess), since the pod didn't exist in the cluster - only in ArgoCD.

> The pod does indeed not exist in the cluster when checking with kubectl get pods.

**Comment by andrii-korotkov-verkada** [comment]
Maybe the cache needs to be invalidated. Normally, ArgoCD should get event from Kubernetes about the deletion and update the cache, but it could be slow on medium+ clusters. ArgoCD version 2.13 added some big optimizations, which help with event processing as well at least for medium clusters. I suggest giving it a shot.

**Reply by BeyondEvil** [reply]
Thanks for the tip. Will try!

**Comment by rumstead** [comment]
The phantom pods are a known bug :(. Here is a pretty detailed issue - https://github.com/argoproj/argo-cd/issues/9226
